modules/http/src/main/java/org/mule/module/http/internal/listener/HttpMessageProcessorTemplate.java

```diff
@@ -192,6 +194,11 @@ private void addThrottlingHeaders(org.mule.module.http.internal.domain.response.
         {
             throttledResponseBuilder.addHeader(throttlingHeaderName, throttlingHeaders.get(throttlingHeaderName));
         }
+
+        for (String header : extraHeaders.keySet() )
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Iterate over the "entrySet" instead of the "keySet". ([squid:S2864](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS2864))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7625&issues=AWiOv-8gotIXnmsU1tFs&open=AWiOv-8gotIXnmsU1tFs)

[ISSUE_KEY]: AWiOv-8gotIXnmsU1tFs